u6-spas/src/views/Converter.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+ <table>
+   <tr>
+     <td><label for="inCon"></label>
+      <input
+        id="inCon"
+        v-model.number="input"
+        type="number"
+        min="-273.15"
+        max="+273.15"
+        step="1"
+      /></td>
+     <td>=</td>
+     <td><span class="output" >{{output.toFixed(2)}}</span></td>
+   </tr>
+   <tr>
+      <td>
+        <select v-model="inputUnit">
+          <option v-for="unit in units" :key="unit.id" :value="unit.id" >{{unit.label}}</option>
+        </select>
+      </td>
+      <td class="fake-link" @click="swap">swap </td>
+      <td>
+        <select v-model="outputUnit">
+          <option v-for="unit in units" :key="unit.id" :value="unit.id" >{{unit.label}}</option>
+        </select>
```

Comment:
นี่เป็นตัวอย่างที่ดีในการพูดคุยเกี่ยวกับข้อกำหนดความเป็นเอกลักษณ์ของ `:key` คุณมี 2 v-fors ที่ทั้งคู่ใช้คีย์เดียวกัน ในชั้นเรียน ฉันกล่าวว่าคีย์ต้องไม่ซ้ำกันทั่วโลก อันที่จริงฉันคิดผิด... และนี่คือตัวอย่างที่คีย์ไม่ขัดแย้งกัน! คีย์ต้องไม่ซ้ำกันก็ต่อเมื่อองค์ประกอบ v-for เป็นพี่น้องกัน...

This is a good example for discussing the uniqueness requirements of `:key`. You have 2 v-fors that both use the same keys. In class I mentioned that the key must be globally unique. Actually I was wrong... and this is an example where the keys do not conflict! Keys must only be unique if the v-for elements are siblings...

**โอเค** (v-fors มีองค์ประกอบหลักต่างกัน):  
**Ok** (the v-fors have different parent elements):
```vue
<select v-model="inputUnit">
  <option v-for="unit in units" :key="unit.id" :value="unit.id">{{ unit.label }}</option>
</select>
<select v-model="outputUnit">
  <option v-for="unit in units" :key="unit.id" :value="unit.id">{{ unit.label }}</option>
</select>
```
**ไม่โอเค** (v-fors มีองค์ประกอบหลักและคีย์เดียวกัน):  
**Not ok** (the v-fors have the same parent element & the same keys):
```vue
<div>
  <div v-for="unit in units" :key="unit.id" :value="unit.id">
    <a :href="`#${unit.label}`">Jump to {{ unit.label }}</a>
  </div>
  <div v-for="unit in units" :key="unit.id" :value="unit.id">
    <h1 :id="unit.id">{{ unit.label }}</h1>
    <p>{{ unit.description }}</p>
  </div>
</div>
```
**แก้ไขสำหรับกรณีที่ 2** (เราให้ v-for แต่ละคำนำหน้าที่ไม่ซ้ำกัน)
**Fix for the 2nd case** (we give each v-for a unique prefix)
```vue
<div>
  <div v-for="unit in units" :key="`toc-${unit.id}`" :value="unit.id">
    <a :href="`#${unit.label}`">Jump to {{ unit.label }}</a>
  </div>
  <div v-for="unit in units" :key="`chapter-${unit.id}`" :value="unit.id">
    <h1 :id="unit.id">{{ unit.label }}</h1>
    <p>{{ unit.description }}</p>
  </div>
</div>
```